### PR TITLE
adding FAQs

### PR DESCRIPTION
**Pull request by robertwhiffin**

### CODE REVIEW

File: FAQs/FAQs.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Is there a performance difference between sparklyr and SparkR?
+to do
+
+## Can Spark make everything faster? How do I parellelise workloads?
+Spark is not magic - taking a long running piece of code and executing it on Spark without any changes will result in a long running piece of code running on Spark. There are 2 things to think about when parallelising workloads: running actual Spark code (through SparkR or sparklyr) or using Spark as a distribution engine for spreading calculations across a cluster. Writing Spark code is already parallelised and can be optimised by the developer, but does not require anything in particular to parallelise code. 
+
+Using Spark as a distribution engine allows developers to parallelise their existing workloads through functions such as `sparklyr::spark_apply()` or `SparkR::gapply()`. These functions allow you to pass vanilla R code for Spark to execute on its worker nodes, effectively running many processes at once. This is known as using a UDF - a User Defined Function.
```
Comment (RafiKurlansik) [inline FAQs/FAQs.md]: Add links to the functions docs.

Comment (RafiKurlansik) [reply FAQs/FAQs.md]: Also add a link to the guide on UDFs in this repo in that last sentence.

File: FAQs/FAQs.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Not all workloads are suitable for this parallelisation however. The characteristic of a task that is well suited to paralellisation is that it can be broken down into independent subtasks. Imagine training a model on a 100 different subsets of data - you could train one model at a time, or 100 all at once and still get the same result. Workloads with complex dependencies are harder to parallelise. Consider any iterative algorithm where the input of task `n` is the result of task `n-1`. Such algorithms require reformulating before they can be parallelised.
+
+## How do I troubleshoot?
+Troubleshooting is most difficult when using Spark as a distribution engine. Because each parallel run happens in an ephemeral R session on the worker nodes, the root cause of an 
+error sometimes fails to make it to the main driver logs, which requires digging into the worker logs to understand why your code is failing. If working in a notebook, the worker logs can be found by navigating to the Executors tab after clicking through to the details of the failed job. If using RStudio, you'll need to navigate to the SparkUI tab of the cluster's page in Databricks and find the executor's logs.
```
Comment (RafiKurlansik) [inline FAQs/FAQs.md:39]: The stack traces for errors with UDFs have gotten a bit better with SparkR.  They will actually show up in the notebook; no need to dig into worker logs thankfully.  Not sure if that holds true for sparklyr.

File: FAQs/FAQs.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+## Best practice - things to try and do
```
Comment (RafiKurlansik) [inline FAQs/FAQs.md:42]: Do all of your data engineering work with SparkR/sparklyr, and only drop data back into an R dataframe when you need to use packages from the R ecosystem.

File: FAQs/FAQs.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- When developing a UDF, start small and continually test. Start with a simple function and add complexity, continually running and testing your function as you go. When you introduce a bug, this will make it easy to know which line of code is responsible, rather than debugging a whole script. 
+- Use native SparkR or sparklyr functionality when possible - it will run faster due to the optimisation engine. Avoid UDFs unless necessary. 
+- If using UDFs, turn off Adaptive Query Execution.
+	- sparklyr: `spark_session_config(sc, "spark.databricks.optimizer.enabled", F)`
+	- SparkR: `sparkR.session(sparkConfig = list("spark.databricks.optimizer.adaptive.enabled" = F))`
+
+## Worst practice - things to try and avoid
+- Collecting large datasets. Do you *really* need all the data on the driver? This loses all the benefits of Spark. A leading cause of Out of Memory errors.
+- Performing aggregations on highly skewed data without
```
Comment (RafiKurlansik) [inline FAQs/FAQs.md]: Sentence fragment?

File: FAQs/FAQs.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+- Lacks full support of Spark MLib - in particular missing the feature engineering functions. 
+
+### sparklyr
+- Good support for data science use cases; full set of feature engineering and distributed ML algorithms. 
+- Compatibility with dplyr syntax reduces learning curve and makes it easy to switch between writing distributed queries on Spark and vanilla queries on R dataframes.
+- Lacks full support of Spark APIs without extension. For example, SparkR's `to_json()` has no equivalent in sparklyr, whilst dealing with nested data requires the package `sparklyr.nested`.
+
+In summary, choosing whether to use sparklyr or SparkR should depend on what you are trying to do and may call for flexibility even within the same project. Both projects are open source allowing users to inspect and contribute to the code base, both have good documentation and user communities for seeking help.
+
+## Is there a performance difference between sparklyr and SparkR?
```
Comment (RafiKurlansik) [inline FAQs/FAQs.md:21]: On Databricks there is an out of the box performance optimization when transferring data between SparkR and R.  This works when collecting data, pushing data to Spark, or running UDFs. 

With sparklyr you can get the same benefits by installing Apache Arrow (link to install guide).

Anecdotally, we have observed some performance issues when using windowing functions with sparklyr compared to SparkR.  We've also observed that UDFs are more scalable and stable with SparkR.

Comment (robertwhiffin) [reply FAQs/FAQs.md:21]: Thanks. I'll have a look at running some tests to quantify those points and see how much of a difference it makes

Comment (RafiKurlansik) [reply FAQs/FAQs.md:21]: It's basically the difference between having Arrow on and not (so, orders of magnitude).  Like 40 seconds vs. 40 minutes.

File: FAQs/FAQs.md (hunk 4 above)
Comment (robertwhiffin) [reply FAQs/FAQs.md]: more of a thought fragment. half thought, half written. Thinking about it further, this was going the lines of "don't call collect_list() on an entire table". will update to reflect

**Pull request merged**